keps/sig-cluster-lifecycle/kubeadm/20190424-kubeadm-for-windows.md

```diff
@@ -0,0 +1,287 @@
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+In Kubernetes 1.14, official support was added for Windows Containers. However, there is no official tool available to users for joining a Windows node to a cluster. The current solution is to create a set of scripts to download Kubernetes binaries and write their config files, but this has proven to be cumbersome and is a huge pain point for Windows adoption. On Linux, Kubeadm is available to quickly join nodes to a cluster, and the intent of this KEP is to propose a design to implement the same functionality for Windows.
+
+It should be noted that this PR only proposes enablement of support for Windows worker nodes using kubeadm.
+
+## Motivation
+
+The motivation for this KEP is to provide a tool to allow users to take a Windows machine and join it to an existing Kubernetes cluster with a single command. The user should also be able to reset the node with a single command. We will also provide the tools necessary to install prerequisites and to bring the Windows node to a Ready state.
+
+### Goals
+
+* Create and maintain a Powershell script to install and run Kubernetes prerequisites on Windows
+* Support kubeadm join for Windows
+* Support kubeadm reset for Windows
+* Create and maintain a Powershell script to deploy Windows CNIs
+
+### Non-Goals
+
+* Installing the Container Runtime (e.g. Docker or containerd)
+* Implement kubeadm init for Windows
+* Implement kubeadm join --control-plane
+* Supporting upgrades using kubeadm upgrade (to be revisited for Beta)
+* Running kube-proxy as a DaemonSet on Windows
+* Running Flannel as a DaemonSet on Windows
```

Comment:
please add `(to be revisited for Beta)` after `Running kube-proxy as a DaemonSet on Windows` and `Running Flannel as a DaemonSet on Windows`.

whether we end up working on that depends on external factors - e.g. Windows core.